alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -44,6 +44,18 @@ AgentId BasicBehaviour::getOwnId() const
     return _engine->getTeamManager().getLocalAgentID();
 }
 
+void BasicBehaviour::stop()
```

Comment:
This is pretty common where the derived class method calls the base class method of the same name. Eg. operator=. I did a small variation by prefixing with`do` just to maintain consistency in the naming with doInit etc.

Reply:
They don't have the same name here right?  The problem is the large amount of functions with similar names:
`doStart` `doInit` `initializeParameters` `initJob` `start` `activate` `_onInit`

It makes it really hard to remember which functions belong to which class and are called at what time.

Reply:
unfortunately the names are part of the external interface because of the auto generated code & we wanted to maintain backward compatibility. For behaviour's we have initialiseParameters, run & onTermination, but for plans we have onInit, run & onTerminate. Further behaviour's have an init() public method as well which I have no clue who uses it so I decided to follow the convention: Runnable object uses - doStart, doStop ... doRun etc. & the derived classes use initJob, runJob etc. & onInit_, onRun_ are wrappers for the auto generated code. @abhi1287 what's your take on this? Personally I think this is a different problem & should be taken up in a different PR.